### PR TITLE
M3-3923 Firewall Rule Table

### PR DESCRIPTION
**Pull request by johnwcallahan**
## Description

_This PR got large with utility functions and tests – sorry!_

This adds the Firewall Rule Table component. Data is mocked.

Screen Shot 2020-02-26 at 5 10 57 PM

Also not working yet (coming in future PRs):

- Action Menu
- “Add new” buttons

The bulk of this work is the utilities needed to determine if a raw rule is one of our custom types. Please check the unit tests to get a feel for how these work.

Play around with the values here to test: https://github.com/linode/manager/pull/6131/files#diff-ada8a997ebe5cc4c81a104a37113e03dR24

### CODE REVIEW

File: packages/manager/src/features/Firewalls/shared.ts [hunk 1]
```diff
@@ -25,115 +29,191 @@ export const firewallOptionItems = [
...
+  if (protocol !== 'TCP') {
+    return undefined;
+  }
+
+  // All predefined Firewalls allow all IPs.
+  if (!allowsAllIPs(addresses)) {
+    return undefined;
+  }
+
+  if (ports === portPresets.ssh) {
```
Comment (Jskobos) [inline packages/manager/src/features/Firewalls/shared.ts]: `switch` here maybe?

File: packages/manager/src/features/Firewalls/FirewallDetail/FirewallRuleTable.tsx [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+const useStyles = makeStyles((theme: Theme) => ({
+  header: {
+    display: 'flex',
+    justifyContent: 'space-between',
+    flexDirection: 'row'
+  }
+}));
+
+type RuleType = 'inbound' | 'outbound';
```
Comment (Jskobos) [inline packages/manager/src/features/Firewalls/FirewallDetail/FirewallRuleTable.tsx]: This naming is a little confusing. `ruleType` below is used both for `inbound | outbound` and for `ssh | dns | etc`

Comment (johnwcallahan) [reply packages/manager/src/features/Firewalls/FirewallDetail/FirewallRuleTable.tsx]: Oh yeah, good point. Addressed.

File: packages/manager/src/features/Firewalls/shared.ts (hunk 1 above)
Comment (johnwcallahan) [reply packages/manager/src/features/Firewalls/shared.ts]: Sure thing. Yeah, it's a little more readable that way.

**Pull request merged**